src/js/test/rpc/server.test.ts

```diff
@@ -262,5 +262,77 @@ describe("Server", function () {
           });
       }
     );
+
+    it("should apply coprocessors to multiples record batches", function () {
+      const { spyGetHandles } = createStubs(sinonInstance);
+      // transform coprocessor function
+      const uppercase = (record) => ({
+        ...record,
+        value: record.value.map((char) => {
+          if (char > 97 && char < 122) {
+            return char - 32;
+          } else {
+            return char;
+          }
+        }),
+      });
+
+      const coprocessors = [1, 2, 3].map((id) =>
+        createHandle({
+          globalId: BigInt(id),
+          apply: (recordBatch) => {
+            const result = new Map();
+            const transformedRecord =
+              // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+              // @ts-ignore
+              recordBatch.map(({ header, records }) => {
+                return {
+                  header,
+                  records: records.map(uppercase),
+                };
+              });
+            result.set("result", transformedRecord);
+            return result;
+          },
+        })
+      );
+
+      spyGetHandles.returns(coprocessors);
+
+      return Promise.all([
+        client.process_batch({
+          // send 101 record batches for apply 1,2,3 coprocessor
+          requests: "1"
+            .repeat(100)
+            .split("1")
```

Comment:
`split` removes the given separator, so you end up with an array of empty strings: [typescript playground](https://www.typescriptlang.org/play?#code/DYUwLgBAzmBOEF4ICICMyB0sQAcQEMwAKVABlIEoModgBLYtZCgKAGMB7AOyg9A2AcA5kRixWLAPSSIAFQ4RQ+WFwgBbDtgj4ARhwCukMAAsQi-FyH78QkABoIbemwDW2vQDczdVcgCiAB74arQgUMgQmigA6saEAORQEAByIADumFIyAPImILBpdFD2ELaQMMpgIAAmEDoAnhDYGh4+QhB5xY4camogXGBJFrV5EGmawLVFEPUG8LT49UKwBlzVGCwQEEA)

Reply:
Yes, but I just need an array with `n` item, doesn't matter the array's value

Reply:
But why do it though? It misled me into thinking it was wrong, and will probably confuse other readers.

Reply:
Solve, I improve the code, add the Array constructor for this iterable